lib/daru/io/exporters/json.rb

```diff
@@ -0,0 +1,403 @@
+        #
+        #   #=>
+        #   # {
+        #   #   "vectors": [
+        #   #     "name",
+        #   #     "age",
+        #   #     "sex"
+        #   #   ],
+        #   #   "index": [
+        #   #     "child",
+        #   #     "dad",
+        #   #     "mom"
+        #   #   ],
+        #   #   "data": [
+        #   #     [
+        #   #       "Jon Snow",
+        #   #       "Rhaegar Targaryen",
+        #   #       "Lyanna Stark"
+        #   #     ],
+        #   #     [
+        #   #       18,
+        #   #       54,
+        #   #       36
+        #   #     ],
+        #   #     [
+        #   #       "Male",
+        #   #       "Male",
+        #   #       "Female"
+        #   #     ]
+        #   #   ]
+        #   # }
+        #
+        # @example Writing to a JSON file with orient: :records
+        #   df = Daru::DataFrame.new(
+        #     [
+        #       {name: 'Jon Snow', age: 18, sex: 'Male'},
+        #       {name: 'Rhaegar Targaryen', age: 54, sex: 'Male'},
+        #       {name: 'Lyanna Stark', age: 36, sex: 'Female'}
+        #     ],
+        #     order: %i[name age sex],
+        #     index: %i[child dad mom]
+        #   )
+        #
+        #   #=> #<Daru::DataFrame(3x3)>
+        #   #            name          age       sex
+        #   # child   Jon Snow         18       Male
+        #   # dad   Rhaegar Ta         54       Male
+        #   # mom   Lyanna Sta         36     Female
+        #
+        #   Daru::IO::Exporters::JSON.new(df, 'filename.json', orient: :records, pretty: true)
+        #
+        #   #=>
+        #   # [
+        #   #   {
+        #   #     "sex": "Male",
+        #   #     "age": 18,
+        #   #     "name": "Jon Snow"
+        #   #   },
+        #   #   {
+        #   #     "sex": "Male",
+        #   #     "age": 54,
+        #   #     "name": "Rhaegar Targaryen"
+        #   #   },
+        #   #   {
+        #   #     "sex": "Female",
+        #   #     "age": 36,
+        #   #     "name": "Lyanna Stark"
+        #   #   }
+        #   # ]
+        #
+        # @example Writing to a JSON file with static nested JsonPaths
+        #   df = Daru::DataFrame.new(
+        #     [
+        #       {name: 'Jon Snow', age: 18, sex: 'Male'},
+        #       {name: 'Rhaegar Targaryen', age: 54, sex: 'Male'},
+        #       {name: 'Lyanna Stark', age: 36, sex: 'Female'}
+        #     ],
+        #     order: %i[name age sex],
+        #     index: %i[child dad mom]
+        #   )
+        #
+        #   #=> #<Daru::DataFrame(3x3)>
+        #   #            name          age       sex
+        #   # child   Jon Snow         18       Male
+        #   # dad   Rhaegar Ta         54       Male
+        #   # mom   Lyanna Sta         36     Female
+        #
+        #   Daru::IO::Exporters::JSON.new(
+        #     df,
+        #     'filename.json',
+        #     orient: :records,
+        #     pretty: true,
+        #     name: '$..specific..name',
```

Comment:
Why `..` here? In JsonPath search, `..` means "then skip any number of levels". For creation, I believe, `.` is appropriate and `..` should be treated as undefined behavior.